### ISSUE TITLE
Progress information always print in new line.

### ISSUE THREAD

**Issue by hidrodixtion** [Homebrew/brew#981]
Progress information (#) should be in one line, but the latest brew always print # in new line.

![screen shot 2016-09-17 at 11 24 31 am](https://cloud.githubusercontent.com/assets/52334/18605939/34cf57ca-7cca-11e6-884e-60200387acf3.png)

### Homebrew version :
```
Homebrew 0.9.9 (git revision a54c; last commit 2016-09-16)
Homebrew/homebrew-core (git revision 968d; last commit 2016-09-16)
```

**Comment by zmwangx** [comment]
Sounds more like a terminal emulator problem. Which terminal emulator is this, and how many columns do you have?  And what happens when you run the following in bash:

```bash
fill_row () { for (( i = 0; i < COLUMNS; i++ )); do echo -n \#; done; }; fill_row; printf '\r'; fill_row
```

I know on the Windows console this would result in two lines because when you get to the end of line your cursor is already on the next, but I've yet to encounter this in a *nix terminal emulator.

**Comment by hidrodixtion** [comment]
I'm using iTerm2, build 3.0.9. This is the output from your script:

![screen shot 2016-09-17 at 8 27 21 pm](https://cloud.githubusercontent.com/assets/52334/18608504/50e9107c-7d15-11e6-8598-321d81a4738b.png)

**Comment by zmwangx** [comment]
And what's the output of `echo $COLUMNS` when you see the original issue?

I would guess you have a curlrc with the `--progress-bar` option turned on, and `COLUMNS` is 78. Basically, your terminal is not wide enough for curl's rather inflexible progress bar (there might be a way to let it be flexible, I don't use this option so I didn't dig into it).

In any case, it's a curl issue, not a brew one, because the progress bar is printed by curl.

**Comment by hidrodixtion** [comment]
you're right, `echo $COLUMNS` print 78. But, I don't know about curlc and I don't even use curl. Maybe it's because the width of the terminal. Thanks for your help.

**Comment by zmwangx** [comment]
`brew` uses curl internally. You should have a curlrc somewhere because I don't think `--progress-bar` is a default option, but no big deal.

Anyway, everything should be fine if you have at least 79 columns.

**Comment by reitermarkus** [comment]
> I don't think --progress-bar is a default option

https://github.com/Homebrew/brew/blob/10f2bc2deabad58c4f6b80481ccc913faeb85510/Library/Homebrew/global.rb#L33

**Comment by zmwangx** [comment]
> It is

Oh I see, I meant it's not a default for curl itself. Apparently I'm overriding Homebrew's options in my curlrc. Thanks for pointing it out though.

**Comment by zmwangx** [comment]
> I'm overriding Homebrew's options in my curlrc

Actually that's weird, command line args should trump curlrc. Anyway, I don't really care.

**Comment by apjanke** [comment]
@hidrodixtion What's your `$TERM` setting?

**Comment by hidrodixtion** [comment]
it prints `xterm-256color`

**Comment by apjanke** [comment]
And is that the Agnoster theme on `zsh` that you're running, or something else? I'll see if I can reproduce.

**Comment by hidrodixtion** [comment]
I'm using zsh with bullet train theme : https://github.com/caiogondim/bullet-train-oh-my-zsh-theme

**Comment by apjanke** [comment]
And, dumb question, but, did you resize your terminal window while the download was happening? Doing that reproduces for me. (Makes sense - `$COLUMNS` is probably captured at the beginning of the download, and I'm not sure how `TRAPWINCH` propagation works here.)

Actually—in fact, just resizing the window to less than 80 columns reproduces this for me, across multiple `brew fetch` invocations. I think curl might be outputting a fixed 80-column progress bar.

<img width="550" alt="screen shot 2016-09-17 at 11 19 40 pm" src="https://cloud.githubusercontent.com/assets/2618447/18612631/3ca62e02-7d2d-11e6-8fdf-7f83a4ce7fd6.png">

If you widen your window to 82 or more columns, does this still happen?

**Comment by hidrodixtion** [comment]
> did you resize your terminal window while the download was happening? 
No, I'm just doing `brew upgrade` and leave the terminal.

> If you widen your window to 82 or more columns, does this still happen?
I've just tried to do `brew install` with column width: 91 and the problem didn't happen.

**Comment by apjanke** [comment]
Haha. Okay, yep, this is busted. Reopening. I'll look in to this.

Please don't get your hopes too high, though. The answer might be just turning off the progress bar when the terminal is narrow, or "sorry, this isn't something we can fix right now; please keep your terminal at least 80 columns wide".

**Comment by apjanke** [comment]
I've put up a fix at #1002 which disables the progress bar when the terminal is too narrow.
